### PR TITLE
Testing: Upgrade test related development dependencies

### PR DESCRIPTION
**Pull request by gziolo**
This PR updates all test runner related development dependencies. There should be no influence on production code.

I also removed two `devDependencies` that doesn't seem to be referenced in the code anymore:
* `blanket`
* `localStorage` (I forgot to remove it when I referenced our own extracted `react-test-env` dependency)

To make it smooth I also had to refactor 2 test files because one was causing errors and another one started to fail on all tests.

### Testing
1. Make sure Eslint checks are green.
2. Make sure all unit tests pass.
3. Make sure Circle CI is still able to read tests output :)
screen shot 2016-10-05 at 15 18 06

### CODE REVIEW

**Comment by blowery** [comment]
`blanket` was mine, fine to remove it.

**Comment by aduth** [comment]
I started a small refactor of the `wrap-es6-functions` tests which turned into a bit of a bigger refactor in ce2f346. Hope that's okay. Otherwise looks good to me 👍

File: client/lib/wp/sync-handler/test/cache-index.js [hunk 1]
```diff
@@ -30,18 +30,13 @@ describe( 'cache-index', () => {
 	describe( '#getAll', () => {
 		beforeEach( clearLocal );
 
-		it( 'should return undefined for empty localforage', ( done ) => {
+		it( 'should return undefined for empty localforage', () => {
 			return cacheIndex.getAll()
 				.then( res => {
-					try {
```
Comment (gziolo) [inline client/lib/wp/sync-handler/test/cache-index.js]: FYI: Mocha wants us to return `Promise` or use `done` with new library versions, we can't use both :)

**Comment by gziolo** [comment]
@aduth Nice improvement :) I tested it using:
`npm run test-client client/lib/wrap-es6-functions/`

Looks good!

File: client/lib/wp/sync-handler/test/cache-index.js (hunk 1 above)
Comment (jhnstn) [inline client/lib/wp/sync-handler/test/cache-index.js]: I was curious about the changes and dug up the changelog. 
https://github.com/mochajs/mocha/blob/master/CHANGELOG.md#300--2016-07-31
looks good

**Comment by gwwar** [comment]
👍 Thanks for taking this on @gziolo! Changes here look good to me

**Pull request merged**